repo/incubating/flink/versions/0/flinkapplication-frameworkversion.yaml

```diff
@@ -0,0 +1,245 @@
+                {{#JAR_PATH}}
+                value: {{JAR_PATH}}
+                {{/JAR_PATH}}
+              - name: JOBMANAGER
+                 {{#DEPLOY_OWN_CLUSTER}}
+                value: {{NAME}}-{{CLUSTER_NAME}}-jobmanager
+                {{/DEPLOY_OWN_CLUSTER}}
+                {{^DEPLOY_OWN_CLUSTER}}
+                value: {{CLUSTER_NAME}}-jobmanager
+                {{/DEPLOY_OWN_CLUSTER}}
+              - name: CONFIGMAP
+                value: {{NAME}}-flink
+              - name: CLASSNAME
+                value: {{CLASSNAME}}
+              - name: PROGRAM_ARGS
+                value: {{JOB_ARGUMENTS}}
+              - name: SAVEPOINT_PATH
+                valueFrom:
+                  configMapKeyRef:
+                    name: {{NAME}}-flink
+                    key: "location"
+              name: {{PLAN_NAME}}
+              image: kudobuilder/flink-submitter:1.7.2
+              imagePullPolicy: Always
+    stop.yaml: |
+      # job to start
+      apiVersion: batch/v1
+      kind: Job
+      metadata:
+        namespace: default
+        name: stop-flink-job
+      spec:
+        template:
+          metadata:
+            name: {{PLAN_NAME}}-job
+          spec:
+            restartPolicy: OnFailure
+             {{#JAR_URL}}
+            initContainers:
+            - name: download
+              # Do some download things here and save to a shared Volume
+              # TODO
+            {{/JAR_URL}}
+            containers:
+            - env:
+              - name: JAR_PATH
+                {{#JAR_URL}}
+                value: /volume/{{NAME}}.jar
+                {{/JAR_URL}}
+                {{#JAR_PATH}}
+                value: {{JAR_PATH}}
+                {{/JAR_PATH}}
+              - name: JOBMANAGER
+                 {{#DEPLOY_OWN_CLUSTER}}
+                value: {{NAME}}-{{CLUSTER_NAME}}-jobmanager
+                {{/DEPLOY_OWN_CLUSTER}}
+                {{^DEPLOY_OWN_CLUSTER}}
+                value: {{CLUSTER_NAME}}-jobmanager
+                {{/DEPLOY_OWN_CLUSTER}}
+              - name: CONFIGMAP
+                value: {{NAME}}-flink
+              - name: CLASSNAME
+                value: {{CLASSNAME}}
+              - name: PROGRAM_ARGS
+                value: {{JOB_ARGUMENTS}}
+              command: ["./shutdown.sh"]
+              name: {{PLAN_NAME}}
+              image: kudobuilder/flink-submitter:1.7.2
+              imagePullPolicy: Always
+
+  tasks:
+    dependencies:
+      resources:
+        - cluster.yaml
+        - config.yaml
+    run:
+      resources:
+        - start.yaml
+    stop:
+      resources:
+        - stop.yaml
+    restart:
+      resources:
+        - restart.yaml
+  plans:
+    deploy:
+      strategy: serial
+      phases:
+        - name: dependencies
+          strategy: serial
+          steps:
+            - name: dependencies
+              tasks:
+                - dependencies
+    run:
+      strategy: serial
+      phases:
+        - name: dependencies
```

Comment:
why is this phase called dependencies when it is calling `run` but no dependencies?